### PR TITLE
[python] Consolidate+vacuum (except bulk array data) on ingest

### PR DESCRIPTION
**Pull request by johnkerl**
@Shelnutt2 I'm looking for your feedback here please

### CODE REVIEW

File: apis/python/src/tiledbsoma/tiledb_array.py [hunk 1]
```diff
@@ -177,3 +177,27 @@ def _chunk_is_contained_in(
...
+        which is more time-consuming and should be done at the user's opt-in discretion.
+        """
+
+        if self.uri.startswith("tiledb://"):
+            # At present we need to look up the S3 path. This can only be done via
+            # tiledb.cloud.info(self.uri). Yet this open-source package does not take a dependency
+            # on the tiledb.cloud package.
+            return
+
+        for mode in ["fragment_meta", "commits", "array_meta"]:
```
Comment (Shelnutt2) [inline apis/python/src/tiledbsoma/tiledb_array.py:194]: I don't think we need to consolidate the `array_meta`. That should only written once per write so there isn't a need to consolidate. Multiple array metadata files will exist only across multiple independent writes so we can keep them for time traveling.

File: apis/python/src/tiledbsoma/tiledb_array.py [hunk 2]
```diff
@@ -177,3 +177,27 @@ def _chunk_is_contained_in(
...
+
+            cfg = self._ctx.config()
+            cfg["sm.mem.total_budget"] = 4 * 1024**3
```
Comment (Shelnutt2) [inline apis/python/src/tiledbsoma/tiledb_array.py:197]: This won't affect the consolidation of `commits` or `fragment_meta`.

File: apis/python/src/tiledbsoma/tiledb_array.py [hunk 3]
```diff
@@ -177,3 +177,27 @@ def _chunk_is_contained_in(
...
+            cfg["sm.consolidation.buffer_size"] = 512 * 1024**2
```
Comment (Shelnutt2) [inline apis/python/src/tiledbsoma/tiledb_array.py:198]: This won't affect the consolidation of `commits` or `fragment_meta`.

File: apis/python/src/tiledbsoma/tiledb_array.py [hunk 4]
```diff
@@ -177,3 +177,27 @@ def _chunk_is_contained_in(
...
         return True
+
+    def _consolidate_and_vacuum_fragment_metadata(self) -> None:
+        """
+        This post-ingestion helper consolidates and vacuums fragment metadata -- this is quick to
+        do, and positively impacts query performance.  It does _not_ consolidate bulk array data,
+        which is more time-consuming and should be done at the user's opt-in discretion.
+        """
+
+        if self.uri.startswith("tiledb://"):
```
Comment (Shelnutt2) [inline apis/python/src/tiledbsoma/tiledb_array.py:188]: Go ahead and remove this. No need to differentiate here.

Comment (johnkerl) [reply apis/python/src/tiledbsoma/tiledb_array.py:188]: @Shelnutt2 

```
>>> soma.obs._consolidate_and_vacuum_fragment_metadata()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/johnkerl/git/single-cell-data/TileDB-SOMA/apis/python/src/tiledbsoma/tiledb_array.py", line 195, in _consolidate_and_vacuum_fragment_metadata
    tiledb.consolidate(self.uri, ctx=ctx)
  File "tiledb/libtiledb.pyx", line 3860, in tiledb.libtiledb.consolidate
  File "tiledb/libtiledb.pyx", line 3947, in tiledb.libtiledb._consolidate_timestamp
  File "tiledb/libtiledb.pyx", line 451, in tiledb.libtiledb._raise_ctx_err
  File "tiledb/libtiledb.pyx", line 436, in tiledb.libtiledb._raise_tiledb_error
tiledb.cc.TileDBError: C API: TileDB Internal, std::exception; Consolidation is not supported for remote arrays.
```

File: apis/python/src/tiledbsoma/tiledb_array.py [hunk 5]
```diff
@@ -177,3 +177,29 @@ def _chunk_is_contained_in(
                 return False
 
         return True
+
+    def _consolidate_and_vacuum_fragment_metadata(self) -> None:
+        """
+        This post-ingestion helper consolidates and vacuums fragment metadata -- this is quick to
```
Comment (aaronwolen) [inline apis/python/src/tiledbsoma/tiledb_array.py:183]: Mention it consolidates/vacuums commit files too?

File: apis/python/src/tiledbsoma/tiledb_array.py (hunk 4 above)
Comment (aaronwolen) [reply apis/python/src/tiledbsoma/tiledb_array.py:188]: @johnkerl, Seth indicated that support for `tiledb://` consolidating/vacuuming is complete and will be rolled out soon, so we can drop this exception.

**Review by aaronwolen — APPROVED** [review]

**Comment by johnkerl** [comment]
[sc-24149]

**Pull request merged**